src/ansible_runner/display_callback/callback/awx_display.py

```diff
@@ -222,31 +230,40 @@ def get_begin_dict(self):
     def get_end_dict(self):
         return {}
 
-    def dump(self, fileobj, data, max_width=78, flush=False):
+    def dump(self, fileobj, data, max_width=78):
         b64data = base64.b64encode(json.dumps(data).encode('utf-8')).decode()
-        with self.display_lock:
-            # pattern corresponding to OutputEventFilter expectation
-            fileobj.write('\x1b[K')
-            for offset in range(0, len(b64data), max_width):
-                chunk = b64data[offset:offset + max_width]
-                escaped_chunk = f'{chunk}\x1b[{len(chunk)}D'
-                fileobj.write(escaped_chunk)
-            fileobj.write('\x1b[K')
-            if flush:
-                fileobj.flush()
+        # pattern corresponding to OutputEventFilter expectation
+        out = '\x1b[K'
+        for offset in range(0, len(b64data), max_width):
+            chunk = b64data[offset:offset + max_width]
+            out += f'{chunk}\x1b[{len(chunk)}D'
+        out += '\x1b[K'
+        fileobj.write(out)
 
     def dump_begin(self, fileobj):
         begin_dict = self.get_begin_dict()
         self.cache.set(f":1:ev-{begin_dict['uuid']}", begin_dict)
         self.dump(fileobj, {'uuid': begin_dict['uuid']})
 
     def dump_end(self, fileobj):
-        self.dump(fileobj, self.get_end_dict(), flush=True)
+        self.dump(fileobj, self.get_end_dict())
 
 
 event_context = EventContext()
 
 
+@functools.cache
+def _getsignature(f: t.Callable) -> inspect.Signature:
+    return inspect.signature(f)
+
+
+@functools.cache
+def _getcallargs(sig: inspect.Signature, *args: P.args, **kwargs: P.kwargs) -> types.MappingProxyType:
+    ba = sig.bind(*args, **kwargs)
+    ba.apply_defaults()
+    return types.MappingProxyType(ba.arguments)
```

Comment:
This stuff is unnecessary for this PR, and if desired I can remove the last commit. Looking at things like this just drove me a little crazy:

```python
log_only = args[5] if len(args) >= 6 else kwargs.get('log_only', False)
```

Reply:
You can't safely `functools.cache` the bound args this way- any non-hashable value passed to a display method would cause a runtime error.  There isn't really a benefit to caching this one in real-world usage anyway, since every unique combination of args is a new cache entry, so this would basically just be a memory leak bound by the size of the cache. Several of the display methods happily accept unhashable args, and last I checked there are a number of callers that blindly pass arbitrary objects to display methods.

Caching the signature this way should be fine (though you might want to verify that `inspect` isn't already doing so), since method objects are usually immutable/hashable.

Reply:
`Signature` objects are hashable, and as far as I have been able to confirm they are not cached within `inspect`.

I was also skeptical of caching the `_getcallargs` function, purely for the concern of how much it would cache.  I'll remove that one.